docs/decisions/0024-multi-modal-requirements.md

```diff
@@ -0,0 +1,296 @@
+not `IChatResult` objects.
+
+#### Ideal developer experience
+
+Ideally, the `XmlPromptParser` class should be in the code path of all return types. This would allow the `FunctionCallingStepwisePlanner` to
+use prompt functions to get the necessary function calls.
+
+#### Potential solution
+
+There are two options where this can happen:
+
+1. In the `InvokeAsync` method of the prompt function
+2. In the connectors
+
+Option #1 was previously not chosen because of the fear that it could not scale to the number of model types. Today, for example, there are 39
+different model types in the HuggingFace library (as well as a custom model type). Most of these model types have different input schema, so
+it's not clear how the `InvokeAsync` method could be implemented in a way that would work for all of them.
+
+By moving the logic to the connector (option #2), each of the 39 model types could have their own implementation that parsed the prompt in the
+way that the model expects with the `XmlPromptParser`.
+
+To enforce #2, each service interface would need to require a method that accepted a single string as input with the correct return type.
+For example, the `IChatCompletion` interface could require the following four methods:
+
+```csharp
+public interface IChatCompletion : IAIService
+{
+    ChatHistory CreateNewChat(string? instructions = null);
+
+    // Existing methods for professional developers
+    Task<IReadOnlyList<IChatResult>> GetChatCompletionsAsync(
+        ChatHistory chat,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+
+    IAsyncEnumerable<IChatStreamingResult> GetStreamingChatCompletionsAsync(
+        ChatHistory chat,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+
+    // New methods for prompt engineers to use with prompt functions
+    Task<IReadOnlyList<IChatResult>> GetChatCompletionsAsync(
+        string prompt,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+
+    IAsyncEnumerable<IChatStreamingResult> GetStreamingChatCompletionsAsync(
+        string prompt,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+}
+```
+
+#### Scoping for V1.0.0
+
+To keep V1.0.0 scoped, we just need to ensure the v1.0.0 interfaces have the methods that require a single string as input so they
+can be called by prompt functions. The interfaces that must be updated are:
+
+- ITextCompletion
+- IChatCompletion
+- And _maybe_ IAIService
+
+_All other services (e.g., IImageGeneration) are marked experimental, so they can be ignored for right now._
+
+### Requirement #3: Make it easy to select a service with a required return type
+
+Today, each of our service interfaces have a unique return type:
+
+- `ITextCompletion`: `ITextResult`
+- `IChatCompletion`: `IChatResult`
+- `IImageGeneration`: `IImageResult`
```

Comment:
We are missing `IImageResult` but we should add this. Created https://github.com/microsoft/semantic-kernel/issues/3768